### ISSUE TITLE
Cypress crashes when using Cypress Dashboard for parallelization

### ISSUE THREAD

**Issue by ddzz** [cypress-io/cypress#15532]
### Current behavior
I'm trying to use Cypress Dashboard to run a test suite in parallel. When I run Cypress 6.7.1 using `cypress run --config-file config/cypress.json --record --key <KEY>` (without the `--parallel` flag), roughly two thirds of my run before I see the following error. I see the same error with the `--parallel` flag.

I'm seeing the same error on three different environments: locally with macOS 10.15.7, CircleCI with Docker, and a Jenkins build that runs with Docker.

```
TypeError: Cannot read property 'tests' of null
    at Object.exports.flattenSuiteIntoRunnables (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/util/tests_utils.js:12:32)
    at ProjectE2E.onTestsReceived [as _recordTests] (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/modes/record.js:727:30)
    at ProjectE2E.<anonymous> (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:280:32)
    at Generator.next (<anonymous>:null:null)
    at /Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:9:71
    at new Promise (<anonymous>:null:null)
    at __awaiter (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:5:12)
    at Object.onTestsReceivedAndMaybeRecord (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:274:63)
    at SocketE2E.<anonymous> (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:222:32)
    at Generator.next (<anonymous>:null:null)
    at /Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:8:71
    at new Promise (<anonymous>:null:null)
    at __awaiter (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:4:12)
    at Socket.<anonymous> (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:221:82)
    at Socket.emit (events.js:315:20)
    at Socket.onevent (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/socket/node_modules/socket.io/dist/socket.js:253:20)
    at Socket._onpacket (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/socket/node_modules/socket.io/dist/socket.js:216:22)
    at /Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/socket/node_modules/socket.io/dist/client.js:205:28
    at processTicksAndRejections (internal/process/task_queues.js:79:11)

TypeError: Cannot read property 'tests' of null
    at Object.exports.flattenSuiteIntoRunnables (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/util/tests_utils.js:12:32)
    at ProjectE2E.onTestsReceived [as _recordTests] (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/modes/record.js:727:30)
    at ProjectE2E.<anonymous> (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:280:32)
    at Generator.next (<anonymous>:null:null)
    at /Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:9:71
    at new Promise (<anonymous>:null:null)
    at __awaiter (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:5:12)
    at Object.onTestsReceivedAndMaybeRecord (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/project-base.js:274:63)
    at SocketE2E.<anonymous> (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:222:32)
    at Generator.next (<anonymous>:null:null)
    at /Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:8:71
    at new Promise (<anonymous>:null:null)
    at __awaiter (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:4:12)
    at Socket.<anonymous> (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/server/lib/socket-base.js:221:82)
    at Socket.emit (events.js:315:20)
    at Socket.onevent (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/socket/node_modules/socket.io/dist/socket.js:253:20)
    at Socket._onpacket (/Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/socket/node_modules/socket.io/dist/socket.js:216:22)
    at /Users/dariuszdzien/Library/Caches/Cypress/6.7.1/Cypress.app/Contents/Resources/app/packages/socket/node_modules/socket.io/dist/client.js:205:28
    at processTicksAndRejections (internal/process/task_queues.js:79:11)

The Test Runner unexpectedly exited via a exit event with signal SIGSEGV
```

I did not have this issue when using the Cypress Dashboard a few months ago with the same test suite using Cypress 4.8.0.

### Desired behavior
Tests pass without failures.

### Test code to reproduce
Our repo is open-source and available at https://github.com/department-of-veterans-affairs/vets-website. Cypress tests are named `*.cypress.spec.js`.

### Versions
Cypress 6.7.1
Browser: Electron 87 (headless)

**Comment by jennifer-shehane** [comment]
Duplicate of https://github.com/cypress-io/cypress/issues/15512 We're working on a patch. In the meantime, you can downgrade to Cypress 6.6.0.
